tools/run-test.py

```diff
@@ -51,21 +67,112 @@ def run_check(runnable):
+    print('Downloading: SonarQube scanner (version %s)...' % version)
+    sq = urllib2.urlopen('https://sonarsource.bintray.com/Distribution/'
+                         'sonar-scanner-cli/sonar-scanner-cli-%s-windows.zip' % version)
+    dl = sq.read()
+    with open('sonar-scanner-cli-%s-windows.zip' % version, 'wb') as f:
         f.write(dl)
+    print('SonarQube scanner download completed! (version %s)' % version)
+
+def unzip(file):
+    zip_ref = zipfile.ZipFile(file, 'r')
+    zip_ref.extractall('.')
+    zip_ref.close()
+    print('Unzip finished! (%s)' % file)
+
+def copy_all_files_from_folder(src, dst):
+    src_files = os.listdir(src)
+    for file_name in src_files:
+        full_file_name = os.path.join(src, file_name)
+        if (os.path.isfile(full_file_name)):
+            shutil.copy(full_file_name, dst)
+    print('Copy finished!')
+
+def start_sq_server(version, system):
+    print('Starting SQ server...')
+    calling = ''
+    if system == 'Windows':
+        calling = 'start sonarqube-%s\\bin\\windows-x86-64\\StartSonar.bat' % version
+    elif system == 'Linux':
+        calling = '/sonarqube-%s/bin/windows-x86-64/StartSonar.bat&' % version
+    os.system(calling)
+    print('SQ server is started!')
+
+def validate_running_of_sq_server(version, number_of_attempts, wait):
+    sleep(float(wait))
+    while number_of_attempts:
+        try:
+            contents = urllib2.urlopen('http://localhost:9000/api/system/ping').read()
+            return True
+        except:
+            print('SonarQube is not started yet, rechecking...' )
+            number_of_attempts -= 1
+    return False
+
+def analyze(scanner_version, project_folder, system):
+    calling = ''
+    cwd = os.getcwd()
+    scanner_location =  ''
+    if system == 'Windows':
+        scanner_location = cwd + '\\sonar-scanner-%s-windows\\bin\\sonar-scanner.bat' % scanner_version
+        calling = 'start ' + scanner_location
+    elif system == 'Linux':
+        scanner_location = cwd + '/sonar-scanner-%s-windows/bin/sonar-scanner.bat' % scanner_version
+        calling = scanner_location + '&'
+    os.chdir(project_folder)
+    os.system(calling)
+    os.chdir('..')
 
 def main(options):
-    # 0) Try to build the plugins with 'build.py'
+    server_version = options.server_version
+    scanner_version = options.scanner_version
+    src_of_the_plugins = options.pluginsfolder
+    src_of_the_project = options.projectsfolder
+    noa = options.numberofattempts
+    wait = options.wait
+    system = platform.system()
+    # 0, a) Try to build the plugins with 'build.py'
+
+    # execfile("build.py")
+
+    # 0, b) download sonar-server'
+
+    download_sq_server(server_version)
 
-    download_sq_server('7.2.1')
     # 1) download sonar-scanner
+
+    download_sq_scanner(scanner_version)
+
     # 2) unzip both server and scanner
+
+    unzip('sonarqube-%s.zip' % server_version)
+    unzip('sonar-scanner-cli-%s-windows.zip' % scanner_version)
```

Comment:
We should not spam the root folder. Please use the `build/test` both for download and unzip.